pkg/logger/logger.go

```diff
@@ -0,0 +1,66 @@
+package logger
+
+import (
+	"flag"
+	"time"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	zapcr "sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/config"
+)
+
+const logTmFmtWithMS = "2006-01-02 15:04:05.000"
+
+var (
+	logLevel          int
+	customTimeEncoder = func(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
+		enc.AppendString(t.Format(logTmFmtWithMS))
+	}
+	customCallerEncoder = func(caller zapcore.EntryCaller, enc zapcore.PrimitiveArrayEncoder) {
+		enc.AppendString(caller.TrimmedPath())
+	}
+)
+
+func init() {
+	flag.IntVar(&logLevel, "log-level", 0, "Use zap-core log system.")
+}
+
+func ZapLogger(cf *config.NSXOperatorConfig) logr.Logger {
```

Comment:
How about change to InitLogger(), we may use other log framework

Reply:
That's a good suggestion. Right now, this func is specific for ZapLogger.  these code are corss ported from main branch. So, I feel that we could keep this func name as it is at this moment until main branch is changed later. We can change in v3.2.1 branch accordingly.